Add style guide and improve documentation (#44)
cisagov/development-guide: project_setup/README.md

# Project Setup #

> We recommend you use a skeleton for all new repositories following the
> directions below.

Reviewer: Less awkward:
```suggestion
We recommend you follow the directions below and use a skeleton for
all new repositories.
```

> For repositories from skeletons, run `pre-commit install` to run linting

Reviewer: ```suggestion
For repositories created from skeletons, run `pre-commit install` to run linting
```

and other tools before committing a faux pas to the repository which
will immediately cause linting failures.

Once you've set up a repository, make sure to enable
[branch protection](branch-protection.md).

> ## Skeleton Tool 💀🛠 ##

Reviewer: Sentence case for headers.

Author: Thanks - I'm going back and checking the existing files and headers for sentence case compliance.

Author: Fixed in https://github.com/cisagov/development-guide/pull/44/commits/f4bbeece384fc59b14ab45a60999c2417ea0d1b1, gotta catch em all!

Add the name and description, set the repository to public, and then the rest
of the options can be skipped.

If everything looks good, publish your new repository to GitHub:

```bash
git push --set-upstream origin develop
```

## Setting up branch protection ##

Once you've made your initial commits, enable [branch protection](branch-protection.md)
to enforce the codeowners approval requirements for pull requests.

## Setting up Coveralls for Python projects ##

The README for your new Python project will be prepared with a Coveralls badge.
To make the badge work properly, you'll need to add a repository secret.

1. Visit [Coveralls](https://coveralls.io/) and go to `Add Repos`.
1. Select your new repository and enable it. This will take you to a
page with `Python set up for Coveralls`. The code block will have an entry for
`repo_token: <token>`.
1. Copy the `repo_token` value.
1. On GitHub, visit your new repository's `Settings -> Secrets` page.
    - Note: If you don't have access to `Settings`, please contact an
    administrator to do this step for you.
1. Add a `New repository secret` and name it `COVERALLS_REPO_TOKEN` with the
value from Coveralls.

> ## Ansible Requirement File Generation Tool 🧻🛠 ##

Reviewer: You are breaking the rule of using sentence case for headers.

Author: Fixed in https://github.com/cisagov/development-guide/pull/44/commits/f4bbeece384fc59b14ab45a60999c2417ea0d1b1 as part of the old header mop-up.

    git push --set-upstream origin develop
```

## Create and publish the GitHub repository ##

> Once the `skeleton` tool has run, the next step is to publish GitHub remote repository.

Reviewer 2: ```suggestion
Once the `skeleton` tool has run, the next step is to publish to a GitHub remote repository.
```

To publish your new repository on GitHub, the remote must already exist.
[Create a new repository](https://github.com/organizations/cisagov/repositories/new)
on GitHub with the same name as your new local repository.  If you do not
have permission, ask an administrator to create it for you.

Add the repository name and description, set the repository to public, and
skip the rest of the options.

Next, publish your new repository to GitHub:

```bash
git push --set-upstream origin develop
```

## Create an initial pull request ##

You probably want to add code, documentation, and other items to your
repository to customize it from the skeleton and make changes.

Create a new branch called `first-commits` and publish it:

```bash
git checkout -b first-commits
git push origin first-commits --set-upstream
```

Edit the existing files and add your own as needed, then commit your changes,

> push them, and create a PR via GitHub or the command-line for your teammates
> to review.
> 
> GitHub only populates its list of status checks once a pull request (PR)
> has been created so checks can run against changes. Status checks are

Reviewer 2: We should expand "PR" where we first use it, not the second place we use it.

```suggestion
push them, and create a pull request (PR) via GitHub or the command-line for
your teammates to review.

GitHub only populates its list of status checks once a PR
has been created so checks can run against changes. Status checks are
```

Author: Good catch!
